Segunda entrega do projeto

milenascc opened the pull request:
Houve tamanho atraso por conta dos conflitos.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pauloborba/ta/26)

<!-- Reviewable:end -->

joaoadherval committed:
Criação da Feature de Consultar Aluno

TMotaBastos committed:
Feature de Editar Conceito

rcalegario committed:
Adicionei feature AddStudents

Commit:
Adição da feature "Adicionar critério"

joaoadherval committed:
Término da feature StudentConsult

TMotaBastos committed:
Corrigindo a feature de Editar Conceito

joaoadherval committed:
Merge pull request #1 from joaoadherval/StudentConsult

Student consult

rcalegario committed:
Coloquei feature AddStudents em ingles

TMotaBastos committed:
Colocando a feature em ingles de Editar Conceito

TMotaBastos committed:
Merge pull request #1 from rcalegario/master

Feature AddStudents

Commit:
Tradução da feature "Adicionar critério" para inglês

TMotaBastos committed:
Ajeitando algumas palavras em inglês

DLRibeiro-zz committed:
Feature to add marks to students criteria

Commit:
Merge pull request #1 from DLRibeiro/AddMark

Feature to add marks to students criteria

Commit:
Merge pull request #1 from joaoadherval/StudentConsult

StudentConsult

Commit:
Adição da feature Notificação

Commit:
Merge branch 'master' of https://github.com/Milechwan/TA

Commit:
adicionado arquivo removido no primeiro commit

joaoadherval committed:
Merge pull request #2 from ArthurLapprand/master

2+(4)>>1

joaoadherval committed:
Merge pull request #3 from Milechwan/master

Feature de notificação

DLRibeiro-zz committed:
Merge pull request #2 from joaoadherval/master

Adição de Features

DLRibeiro-zz committed:
Merge pull request #1 from TMotaBastos/master

Cenários de Editar Conceito e Adicionar Estudante

Commit:
Correções:
- parametrização do nome da página em GUI Scenarios
- troca de localização do AND

Commit:
Notificação atualizada com base na revisão

joaoadherval committed:
StudentConsult.feature agora está correta de acordo com o que foi corrigdo pelo aluno tmb2

DLRibeiro-zz committed:
Update and rename AddMark.feature to AddEvaluation.feature

DLRibeiro-zz committed:
Mudei pra evaluation

DLRibeiro-zz committed:
Mudei pra evaluation

DLRibeiro-zz committed:
Mudei pra evaluation
Add Evaluation Feature

Commit:
Tradução do nome da feature.

DLRibeiro-zz committed:
Muita coisa

DLRibeiro-zz committed:
Muita coisa

DLRibeiro-zz committed:
Muita coisa

DLRibeiro-zz committed:
Muita coisa

DLRibeiro-zz committed:
Muita coisa

DLRibeiro-zz committed:
Muita coisa

DLRibeiro-zz committed:
Muita coisa

rcalegario committed:
Esquele do Steps de AddStudentes criado

Commit:
isn't -> is not

TMotaBastos committed:
Corrigindo os problemas vistos na correção

rcalegario committed:
Steps de AddStudentes implementados

Commit:
Scenarios de controllers feitos

Commit:
Mudando de volta removendo as aspas.

Commit:
Pequenas mudanças

Commit:
Adicionado scenario de GUI

TMotaBastos committed:
Codigo de implementação de um teste de controlador e um teste de GUI

Commit:
Notificação atualizada para correção de erros de linguagem e criação do teste

rcalegario committed:
Merge pull request #1 from ArthurLapprand/master

Pull request dos Testes

DLRibeiro-zz committed:
Merge pull request #3 from Milechwan/master

Pull request para segunda entrega - testes

DLRibeiro-zz committed:
Merge pull request #4 from DLRibeiro/TesteFeature

Teste feature

Commit:
Merge pull request #2 from rcalegario/master

Pull Request Testes

joaoadherval committed:
Testes da Feature StudentConsult criada

joaoadherval committed:
Merge pull request #4 from TMotaBastos/master

Pull Request da segunda entrega

TMotaBastos committed:
Merge pull request #3 from joaoadherval/master

Criação dos Testes

DLRibeiro-zz committed:
Merge pull request #5 from pauloborba/master

Recreate some files

TMotaBastos committed:
Merge branch 'master' of https://github.com/DLRibeiro/TA into DLRibeiro-master

merge do pull request

DLRibeiro-zz committed:
Ajustei minha feature

Commit:
Commit de todos os arquivos alterados para resolução de conflito

DLRibeiro-zz committed:
Ajustei minha feature

Commit:
Commit de todos os arquivos alterados para resolução de conflito (quarta vez)

DLRibeiro-zz commented on test/cucumber/steps/AddStudentsSteps.groovy:
```diff
@@ -0,0 +1,75 @@
+}
+
+Then (~'^ the student "([^"]*)" with login "([^"]*)" is registered in the system$'){
+    String name, login ->
```
Possível fazer em um único passo. Só utilizando o Student.findByLogin e assert != null

rcalegario commented on test/cucumber/steps/StudentConsultSteps.groovy:
```diff
@@ -0,0 +1,52 @@
+Then(~'^the system will return the information about "([^"]*)"$'){
+    String nome ->
+        estudante  = Student.findByName(nome)
+        assert estudante.StudentConsultTestDataAndOperations.compatibleTo(estudante, nome)
```
StudentConsultTestDataAndOperations é uma classe, e não um método da classe Student

rcalegario commented on test/cucumber/steps/StudentConsultSteps.groovy:
```diff
@@ -0,0 +1,52 @@
+
+When(~'^I search for "([^"]*)"$'){
+    String nome, login ->
+        Student.findByName(nome)
```
Deveria confirmar se o retorno do aluno ocorreu com findByName e é interessante verificar se o aluno retornado é o aluno desejado, para isso seria necessário receber mais informações do aluno no When, com por exemplo o login.

TMotaBastos commented on test/cucumber/steps/ReportsTester.groovy:
```diff
@@ -0,0 +1,51 @@
+        assert ReportsDataAndOperations.compatibleTo(relat1,report1)
+        assert ReportsDataAndOperations.compatibleTo(relat2,report2)
+}
+When(~'^I add the evaluation "([^"]*)" to the student "([^"]*)", login "([^"]*)"$'){String arg, String loginA->
```
O texto contém 3 variáveis, porém só recebe 2 variáveis.

TMotaBastos commented on test/cucumber/steps/ReportsTester.groovy:
```diff
@@ -0,0 +1,51 @@
+
+//Cenários de GUI
+//Ser notificado sobre os problemas de desempenho dos alunos
+Given(~'^there is the performance problem "([^"]*)" with the students "([^"]*)" and "([^"]*)"$'){
```
O mesmo problema de ter 3 variáveis e só recebe 1 variável.

TMotaBastos commented on test/cucumber/steps/ReportsTester.groovy:
```diff
@@ -0,0 +1,51 @@
+           (ReportsDataAndOperations.countType(evalType)/ReportsDataAndOperations.count())*100
+       if(totalE>=70){
+           return true
+       }
```
Acho que tudo isso poderia ser um método do controlador.

TMotaBastos commented on test/cucumber/steps/ReportsTester.groovy:
```diff
@@ -0,0 +1,51 @@
+/**
+ * Created by Milena Carneiro on 28/04/2016.
```
(OK) - Test behavior should strictly conform to scenario semantics
(OK) - Avoid ambiguities due to similar step sentences
(OK) - Do not duplicate test code
(OK) - Test should clean up environment at the end
(OK) - Do not mix GUI and controller tests
(OK) - Tests should be platform and language independent

lapprand commented on test/cucumber/steps/EditarConceito.groovy:
```diff
@@ -0,0 +1,82 @@
+import pages.StudentPages.CreateStudentPage
```
Alguns checks estão OK pois não foram implementadas as classes necessárias para avaliá-los.

(OK) - Test behavior should strictly conform to scenario semantics
(OK) - Avoid ambiguities due to similar step sentences
(OK) - Do not duplicate test code
(OK) - Test should clean up environment at the end
(OK) - Do not mix GUI and controller tests
(OK) - Tests should be platform and language independent

milenascc commented on test/cucumber/steps/AddCriterionsSteps.groovy:
```diff
@@ -0,0 +1,107 @@
+Then system does nothing
+*/
+Given(~'^the criterion named "([^"]*)" already exists on the system$') {
+    String name -> assert Criterions.findByName(name) != null
```
Não que isso esteja errado mas eu não usaria por conta do risco de se comparar com um valor diferente do desejado, mesmo não sendo null

milenascc commented on test/cucumber/steps/AddCriterionsSteps.groovy:
```diff
@@ -0,0 +1,107 @@
+    And I finalize the criterion registration
+    Then I should see the new criterion available on the criteria list
+*/
+
```
Baseado no checklist visto em aula, não há duplicação de código, não há ambiguidades, não se misturou cenário de GUI com de controlador, os testes são independentes de plataforma e está conforme às semânticas dos cenários. Quanto à limpar tudo o que foi feito no final dos testes, isso não é mostrado porque ainda não foram implementadas as classes básicas e/ou auxiliares (ou as que existem não tem seus métodos implementados ainda)

joaoadherval commented on test/cucumber/steps/AddEvaluation.groovy:
```diff
@@ -0,0 +1,178 @@
+package steps
+
```
(OK) - Test behavior should strictly conform to scenario semantics
(OK) - Avoid ambiguities due to similar step sentences
(OK) - Do not duplicate test code
(OK) - Test should clean up environment at the end
(OK) - Do not mix GUI and controller tests
(OK) - Tests should be platform and language independent

Commit:
Adicionando controllers

Commit:
Adicionado CreateCriterionPage e outras pequenas mudanças

Commit:
Modificações na classe de teste e na feature para tentar rodar os testes

DLRibeiro-zz committed:
Merge pull request #6 from Milechwan/master

Só atualizando tudo pro meu projeto

Commit:
Mudanças feitas na monitoria 12/05

Commit:
pequenas mudanças

Commit:
Troca de findBy no Steps por método de CriterionTestDataAndOperations que chama método do CriterionController

DLRibeiro-zz committed:
Merge remote-tracking branch 'origin/master'

DLRibeiro-zz committed:
Ajustei minhas classes dominio

DLRibeiro-zz committed:
Ajustei alguns métodos.

DLRibeiro-zz committed:
Update StudentController.groovy

joaoadherval committed:
Merge pull request #5 from Milechwan/master

Atualização

DLRibeiro-zz committed:
Novas domains e view e controller associado

DLRibeiro-zz committed:
Merge remote-tracking branch 'origin/master'

DLRibeiro-zz committed:
Algumas mudanças por causa de closures

DLRibeiro-zz committed:
FormattedDate no StudentController

DLRibeiro-zz committed:
FormattedDate no StudentController

DLRibeiro-zz committed:
Novas operações pros testes

DLRibeiro-zz committed:
Novas operações pros testes

DLRibeiro-zz committed:
Mudanças em muita coisa

DLRibeiro-zz committed:
Mudanças em muita coisa

Commit:
Recriação do CriterionController

Commit:
Correção de bug do IntelliJ

Commit:
Tentando ajeitar mais bugs do IntelliJ

Commit:
Tentando corrigir bug do IntelliJ parte 2

Commit:
Corrigindo Bugs IntelliJ parte 3

Commit:
Acoplamento de métodos no controlador

DLRibeiro-zz committed:
Novo método de controller, e novos construtores para classes básicas

DLRibeiro-zz committed:
Novo método de controller, e novos construtores para classes básicas

Commit:
2 testes passam, 1 falha

joaoadherval committed:
Testes da Feature StudentConsult criada

joaoadherval committed:
Test's functions implemented, looking to test it. Can't merge with atualizations

DLRibeiro-zz committed:
Alterei minha feature, não consegui terminar tudo

Commit:
Modificações na classe de teste , na feature, controladores e testanddataoperations

Commit:
2 testes passam, 1 falha - tentando configurar o chromedriver

Commit:
Commit para resolver conflitos

Commit:
Merge branch 'master' of https://github.com/DLRibeiro/TA into DLRibeiro-master

# Conflicts:
#	grails-app/domain/ta/Evaluation.groovy
#	grails-app/domain/ta/Student.groovy
#	test/functional/steps/EvaluationdDataAndOperations.groovy

Commit:
Merge branch 'DLRibeiro-master'

# Conflicts:
#	grails-app/controllers/ta/CriterionController.groovy
#	grails-app/controllers/ta/StudentController.groovy
#	grails-app/domain/ta/Criterion.groovy
#	grails-app/domain/ta/EvaluationsByCriterion.groovy

Commit:
Merge branch 'refs/heads/pr/2'

# Conflicts:
#	grails-app/controllers/ta/CriterionController.groovy
#	grails-app/controllers/ta/StudentController.groovy
#	grails-app/domain/ta/Criterion.groovy
#	grails-app/domain/ta/Student.groovy
#	grails-app/views/criterion/_form.gsp
#	grails-app/views/criterion/index.gsp
#	grails-app/views/criterion/show.gsp
#	grails-app/views/student/index.gsp
#	test/cucumber/steps/EvaluateStudentSteps.groovy
#	test/functional/steps/EvaluateStudentTestDataAndOperations.groovy

Commit:
2 testes passam, 1 falha - tentando configurar o chromedriver

Commit:
Merge branch 'master' of https://github.com/ArthurLapprand/TA

DLRibeiro-zz committed:
Setei as variaveis iniciais agora

Commit:
Resolvendo conflitos com master de João

Commit:
merge

Commit:
Merge pull request #3 from DLRibeiro/master

Setei as variaveis iniciais agora

Commit:
Adições de métodos e reparação na classe de testes

Commit:
Merge para pull request

rcalegario committed:
Testes funcionando, só com os meus arquivos (sem nenhum pull request"

rcalegario committed:
Modificacao no gitignore

rcalegario committed:
Adicionando tudo de AddStudent

Commit:
Sem GebConfig

rcalegario committed:
merge com milena

joaoadherval committed:
Efetuação dos testes de controller bem sucedidas

rcalegario committed:
view search student with login

Commit:
transferência do projetoToy pra cá

Commit:
adição de unversioned files no gitignore

joaoadherval committed:
Merge branch 'master' of https://github.com/rcalegario/TA into rcalegario-master

Conflicts:
	.idea/modules.xml
	grails-app/controllers/ta/CriterionController.groovy
	grails-app/controllers/ta/StudentController.groovy
	grails-app/domain/ta/Criterion.groovy
	grails-app/domain/ta/EvaluationsByCriterion.groovy
	grails-app/domain/ta/Student.groovy
	grails-app/views/criterion/index.gsp
	grails-app/views/criterion/show.gsp
	grails-app/views/student/index.gsp
	test/cucumber/EvaluateStudent.feature
	test/cucumber/RegisterEvaluation.feature
	test/cucumber/StudentConsult.feature
	test/cucumber/steps/EvaluateStudentSteps.groovy
	test/cucumber/steps/RegisterEvaluation.groovy
	test/cucumber/steps/StudentConsultSteps.groovy
	test/functional/steps/StudentConsultTestDataAndOperations.groovy

Commit:
inspeção de código

Commit:
complementação de testes

Commit:
Criação de views, modificação no controller e na domain class de Report

joaoadherval committed:
Erro nos testes. Botao da View de Search agora funciona. Falta listar alunos

rcalegario committed:
Textinho do commit

Commit:
-added single student average calculation method
-fixed single student average shown when importing group of students

joaoadherval committed:
View search funcionando. Consulta por login apresenta os resultados e trata erro de estudante não existente

joaoadherval committed:
Search funcionando com nomes. Falta fazer com que funcione com os dois, nome e login

joaoadherval committed:
Search funcionando com nomes e logins

joaoadherval committed:
Commit sem pastas bugadas yaay

joaoadherval committed:
Atualizando gitignore

rcalegario committed:
modificacao em um .idea

rcalegario committed:
merge com joao

rcalegario committed:
finalizando merge com joao

Commit:
-bug -> Student not listing correctly

Commit:
-gitignore corrected

Commit:
Merge com Calega

Commit:
Merge branch 'rcalegario-master'

Commit:
-fixed Student criteria and average listing
-fixed searchStudent and importStudents html and css design
-fixed various other controller issues such as refactoring attribute names

Commit:
Commit para dar merge com Lapprand

Commit:
Commit para dar merge com Lapprand

Commit:
Commit para dar merge com Lapprand

Commit:
Funcionamento da feature de Notificação ok (relatórios atualizam ao se adicionar um Evaluation)

Commit:
Modificação do teste (deu um erro muito estranho mas disse que o teste passou)

Commit:
Modificação do teste e de algumas classes (mas ele não passa)

Commit:
Modificação na feature de Notificação, modificações finais em domain class e controller de Report, adição de pages para teste, modificação na classe de ReportTestDataAndOperations

Commit:
Mandando o ReportsDataAndOperations para dar merge com Calegario

Commit:
Mais algumas modificações para dar merge com Calegario (o Report não aceitava mais passar um valor do tipo 0,7 como parâmetro, então outras classes, incluindo as de teste, foram modificadas)

pauloborba commented:
Review status: 0 of 122 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_[.gitignore, line 38 [r1]](https://reviewable.io:443/reviews/pauloborba/ta/26#-KQkZKxxglCDSWRWAksQ:-KQkZKxyhZA04UsqtiPY:b410hbl) ([raw file](https://github.com/pauloborba/ta/blob/acb2a41cf4b0eafa0690c3089baf01c5ad4213b7/.gitignore#L38)):_

> ```
> bin/
> tmp/
> *.tmp
> ```

por que decidiu remover?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pauloborba/ta/26)_

<!-- Sent from Reviewable.io -->
